sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -215,7 +215,25 @@ public synchronized void update(Protos.TaskStatus status) {
         updateStatus();
     }
 
+    /**
+     * WARNING: This is a private method for a reason.  Callers must validate that the taskID is already present in the
+     * tasks map.
+     */
+    private String getTaskName(Protos.TaskID taskID) {
+        return tasks.get(taskID).getTaskInfo().getName();
+    }
+
     private void setTaskStatus(Protos.TaskID taskID, Status status) {
+        GoalStateOverride.Status overrideStatus = stateStore.fetchGoalOverrideStatus(getTaskName(taskID));
```

Comment:
This method now sets both the regular task status and the overridden task status. Does this mean that they need to be kept in sync in some way? If this is the case, how do we ensure this?

Would it make sense to break the override-specific code below into its own method to improve the readability of this method a little. (`updateOverrideStatusIfComplete`?)

Reply:
Factored it out into a different method.  This method is the mechanism of synchronization for the two statuses.  Plan execution drives tasks to their goal state (override or otherwise).  The setting of a Step's status is the canonical source of truth for plan execution.  Plan execution is therefore the correct place to record override status.